lib/manageiq/api/client/collection.rb

```diff
@@ -0,0 +1,52 @@
+require_relative "mixins/action_mixin"
+
+module ManageIQ
+  module API
+    module Client
+      class Collection
+        include ActionMixin
+
+        attr_accessor :server
+
+        def initialize(server)
+          @server = server
+        end
+
+        def search(options = {})
+          options[:expand] = (String(options[:expand]).split(",") | %w(resources)).join(",")
+          result_hash = server.get(name, options)
+          fetch_actions(result_hash)
+          klass = ManageIQ::API::Client::Resource.subclass(name)
+          result_hash["resources"].collect do |resource_hash|
+            klass.new(self, resource_hash)
+          end
+        end
+
+        def self.subclass(name)
+          klass_name = name.camelize
+
+          if ManageIQ::API::Client::Collection.const_defined?(klass_name)
+            ManageIQ::API::Client::Collection.const_get(klass_name)
+          else
+            klass = Class.new(ManageIQ::API::Client::Collection) do
```

Comment:
If you are subclassing Collection, perhaps it's better to use put all the attributes directly on collection...then no need to redefine initialize here.  Can someone do a  base `Collection.new` anyway?